dubbo-configcenter/dubbo-configcenter-apollo/src/main/java/org/apache/dubbo/configcenter/support/apollo/ApolloDynamicConfiguration.java

```diff
@@ -130,13 +135,32 @@ public void removeListener(String key, String group, ConfigurationListener liste
      */
     @Override
     public String getConfig(String key, String group, long timeout) throws IllegalStateException {
-        if (StringUtils.isNotEmpty(group) && !url.getParameter(Constants.CONFIG_GROUP_KEY, DEFAULT_GROUP).equals(group)) {
-            Config config = ConfigService.getAppConfig();
-            return config.getProperty(key, null);
+        if (StringUtils.isNotEmpty(group)) {
+            if (group.equals(url.getParameter(Constants.APPLICATION_KEY))) {
+                return ConfigService.getAppConfig().getProperty(key, null);
+            } else {
+                ConfigService.getConfig(group).getProperty(key, null);
```

Comment:
we need to add return statement here?

Reply:
This is a crucial bug.

Reply:
done

Reply:
Thanks for addressing this. Would it be possible to write a UT for the same? If UT was there, it could have caught already. What do you say?